### PR TITLE
Move conditional queries into model 

### PR DESCRIPTION
**Pull request by gidsg**
### Context
See refactoring style described in http://brendankemp.com/essays/build-a-query-in-activerecord-from-optional-params-the-non-hideous-version/

### Changes proposed in this pull request
Use the chaining ability of scope to handle optional parameters. This removes a lot of conditional logic from the controller and makes the query more readable. 

### Guidance to review
This is a refactor, check show and updates page work the same

### CODE REVIEW

File: app/controllers/registers_controller.rb [hunk 1]
```diff
@@ -31,21 +31,13 @@ def recover_records(fields, params)
 
     sort_by = params[:sort_by] ||= default_sort_by.call
     sort_direction = params[:sort_direction] ||= 'asc'
```
Comment (gidsg) [inline app/controllers/registers_controller.rb]: would be possible to put these default assignments into the model as well if that is preferable.

File: app/models/record.rb [hunk 2]
```diff
@@ -3,4 +3,18 @@ class Record < ApplicationRecord
   belongs_to :register
   scope :current, -> { where("data->> 'end-date' is null") }
   scope :archived, -> { where("data->> 'end-date' is not null") }
+  scope :status, lambda { |status|
```
Comment (danmitchell-) [inline app/models/record.rb]: Does this need to be a lambda?

Comment (gidsg) [reply app/models/record.rb]: this style is preferred by Rubocop for multi-line functions. http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/Lambda

File: app/models/concerns/search_scope.rb [hunk 3]
```diff
@@ -2,11 +2,13 @@ module SearchScope
   extend ActiveSupport::Concern
   included do
     scope :search_for, lambda { |fields, search_term|
-      where(
-        fields
-        .map { |f| "data->> '#{f}' ilike '%#{search_term}%'" }
-        .join(' or ')
-      )
+      if search_term.present?
```
[the remark this answers is not in the document]
Comment (gidsg) [reply app/models/concerns/search_scope.rb]: it does not need to know about `params` directly, only its arguments. If we move the if statement out we have to duplicate this check in both usages of search.

File: app/controllers/registers_controller.rb [hunk 4]
```diff
@@ -31,21 +31,13 @@ def recover_records(fields, params)
...
-              query.current
-            end
-
-    if params[:q].present?
-      query = query.search_for(fields, params[:q])
-    end
-
-    query.order("data->> '#{sort_by}' #{sort_direction.upcase} nulls last").page(params[:page]).per(100)
+
+    @register.records
```
[the remark this answers is not in the document]
Comment (gidsg) [reply app/controllers/registers_controller.rb]: see https://github.com/openregister/registers-frontend/pull/219#discussion_r172192159

File: app/models/concerns/search_scope.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (danmitchell-) [reply app/models/concerns/search_scope.rb]: No problem. I'd just prefer to not hit that concern unless its actually going to be used but i'm not that precious over it. Personal choice.

**Pull request merged**